src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1610,12 +1624,21 @@ class Simplex_tree {
       if (dim == 0) return;
       // Find the maximum filtration value in the border
       Boundary_simplex_range&& boundary = boundary_simplex_range(sh);
-      Boundary_simplex_iterator max_border = std::max_element(std::begin(boundary), std::end(boundary),
-                                                              [](Simplex_handle sh1, Simplex_handle sh2) {
-                                                                return filtration(sh1) < filtration(sh2);
-                                                              });
+      Filtration_value max_filt_border_value;
+      if constexpr (SimplexTreeOptions::is_multi_parameter) {
+        // in that case, we assume that Filtration_value has a `push_to` member to handle this.
```

Comment:
This should also be documented in the concept file `gudhi-devel/src/Simplex_tree/concept/FiltrationValue.h`

Reply:
I've added a few lines in the concept file.